counsel.el

```diff
@@ -2707,7 +2736,17 @@ INITIAL-DIRECTORY, if non-nil, is used as the root directory for search.
 EXTRA-RG-ARGS string, if non-nil, is appended to `counsel-rg-base-command'.
 RG-PROMPT, if non-nil, is passed as `ivy-read' prompt argument."
   (interactive)
-  (let ((counsel-ag-base-command counsel-rg-base-command))
+  (when (eq 'auto counsel-rg-pcre-flag)
+    (setq counsel-rg-pcre-flag
```

Comment:
The values of user options should not be set by programs. It is possible to write this so as to avoid that, I think:
```diff
diff --git a/counsel.el b/counsel.el
index fa214c4..9ee8a20 100644
--- a/counsel.el
+++ b/counsel.el
@@ -2699,6 +2699,15 @@ 'counsel-rg
 (ivy-set-occur 'counsel-rg 'counsel-ag-occur)
 (ivy-set-display-transformer 'counsel-rg 'counsel-git-grep-transformer)
 
+(defun counsel--rg-version ()
+  "Return Ripgrep version string or \"0.0.0\" on failure."
+  (let* ((rg (substring counsel-rg-base-command
+                        0 (string-match-p " " counsel-rg-base-command)))
+         (ver (ignore-errors (counsel--call rg "--version"))))
+    (if (and ver (string-match "\\`ripgrep \\([.[:digit:]]+\\)" ver))
+        (match-string 1 ver)
+      "0.0.0")))
+
 ;;;###autoload
 (defun counsel-rg (&optional initial-input initial-directory extra-rg-args rg-prompt)
   "Grep for a string in the current directory using rg.
@@ -2707,7 +2716,13 @@ counsel-rg
 EXTRA-RG-ARGS string, if non-nil, is appended to `counsel-rg-base-command'.
 RG-PROMPT, if non-nil, is passed as `ivy-read' prompt argument."
   (interactive)
-  (let ((counsel-ag-base-command counsel-rg-base-command))
+  (let ((counsel-ag-base-command counsel-rg-base-command)
+        (counsel--grep-tool-look-around
+         (or (and (eq counsel-rg-pcre-flag 'auto)
+                  (version<= "0.10.0" (counsel--rg-version))
+                  "--pcre2")
+             (string-match-p " \\(?:--pcre2\\_>\\|-[[:alnum:]]*P\\)"
+                             counsel-rg-base-command))))
     (counsel-ag initial-input initial-directory extra-rg-args rg-prompt)))
 (cl-pushnew 'counsel-rg ivy-highlight-grep-commands)
 ```
Disclaimer: 100% untested.

Reply:
Thanks for the code snippet. I have applied your feedback.